x-pack/plugins/fleet/server/services/managed_package_policies.ts

```diff
@@ -26,78 +30,86 @@ export interface UpgradeManagedPackagePoliciesResult {
  */
 export const upgradeManagedPackagePolicies = async (
   soClient: SavedObjectsClientContract,
-  esClient: ElasticsearchClient,
-  packagePolicyIds: string[]
+  esClient: ElasticsearchClient
 ): Promise<UpgradeManagedPackagePoliciesResult[]> => {
   const results: UpgradeManagedPackagePoliciesResult[] = [];
 
-  for (const packagePolicyId of packagePolicyIds) {
-    const packagePolicy = await packagePolicyService.get(soClient, packagePolicyId);
-
-    if (!packagePolicy || !packagePolicy.package) {
-      continue;
-    }
-
-    const installedPackage = await getInstallation({
-      savedObjectsClient: soClient,
-      pkgName: packagePolicy.package.name,
-    });
+  const installedPackages = await getInstallations(soClient, {
+    filter: `${PACKAGES_SAVED_OBJECT_TYPE}.attributes.install_status:installed`,
+  });
 
-    if (!installedPackage) {
-      results.push({
-        packagePolicyId,
-        errors: [`${packagePolicy.package.name} is not installed`],
-      });
-
-      continue;
-    }
-
-    if (shouldUpgradePolicies(packagePolicy.package.version, installedPackage)) {
-      // Since upgrades don't report diffs/errors, we need to perform a dry run first in order
-      // to notify the user of any granular policy upgrade errors that occur during Fleet's
-      // preconfiguration check
-      const dryRunResults = await packagePolicyService.getUpgradeDryRunDiff(
+  for (const { attributes: installedPackage } of installedPackages.saved_objects) {
+    if (installedPackage.keep_policies_up_to_date) {
```

Comment:
Could we move this condition into the `filter` on `getInstallations`?

Reply:
done